packages/core/src/feature-app-manager.ts

```diff
@@ -45,8 +45,11 @@ export interface FeatureAppEnvironment<
    * If this callback is defined, a short-lived Feature App can call this
    * function when it has completed its task. The Integrator (or parent Feature
    * App) can then decide to e.g. unmount the Feature App.
+   *
+   * By default it is assumed that the FA exited successfully, but it can signal
+   * that it failed. It is up to the integrator to handle this.
    */
-  readonly done?: () => void;
+  readonly done?: (failed?: boolean) => void;
```

Comment:
For scalability/flexibility reasons I would prefer to use an unknown result parameter instead, i.e.:

```suggestion
  readonly done?: (result?: unknown) => void;
```

The type of the `result` can then either be casted, or type predicates might be used to [narrow](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) the type at runtime.

Reply:
I am not sure, initially i was thinking about a more elaborate parameter, but this would tie integrator and FA more tightly than i care about.
So maybe `"ok" | unknown` with unknown always being an error indicator?